runtime/neumann/src/lib.rs

```diff
@@ -1151,10 +1151,64 @@ impl_runtime_apis! {
 		}
 	}
 
-	impl pallet_automation_time_rpc_runtime_api::AutomationTimeApi<Block, AccountId, Hash> for Runtime {
+	impl pallet_automation_time_rpc_runtime_api::AutomationTimeApi<Block, AccountId, Hash, Balance> for Runtime {
 		fn generate_task_id(account_id: AccountId, provided_id: Vec<u8>) -> Hash {
 			AutomationTime::generate_task_id(account_id, provided_id)
 		}
+		/**
+		 * The get_time_automation_fees RPC function is used to get the execution fee of scheduling a time-automation task.
+		 * This function requires the action type and the number of executions in order to generate an estimate.
+		 * However, the AutomationTime::calculate_execution_fee requires an Action enum from the automation time pallet,
+		 * which requires more information than is necessary for this calculation.
+		 * Therefore, for ease of use, this function will just require an integer representing the action type and an integer
+		 * representing the number of executions. For all of the extraneous information, the function will provide faux inputs for it.
+		 *
+		 * Action enums:
+		 * - 0: Notify
+		 * - 1: NativeTransfer
+		 * - 2: XCMP
+		 */
+		fn get_time_automation_fees(
+			action: u8,
+			executions: u32,
+		) -> Balance {
+			match action {
+				0 => {
+					let action_enum = pallet_automation_time::Action::Notify {
+						message: "get_fees".into()
+					};
+					match AutomationTime::calculate_execution_fee(&action_enum, executions) {
+						Ok(balance) => balance,
+						Err(_e) => 0,
+					}
+				},
+				1 => {
+					let action_enum = pallet_automation_time::Action::NativeTransfer {
+						sender: sp_runtime::AccountId32::new([1u8; 32]),
+						recipient: sp_runtime::AccountId32::new([2u8; 32]),
+						amount: 1_000_000,
+					};
+					match AutomationTime::calculate_execution_fee(&action_enum, executions) {
+						Ok(balance) => balance,
+						Err(_e) => 0,
+					}
+				},
+				2 => {
+					let action_enum = pallet_automation_time::Action::XCMP {
+						para_id: cumulus_primitives_core::ParaId::from(2114),
+						call: vec![0, 1],
+						weight_at_most: 1_000_000_000,
+					};
+					match AutomationTime::calculate_execution_fee(&action_enum, executions) {
+						Ok(balance) => balance,
+						Err(_e) => 0,
+					}
+				},
```

Comment:
Do we need to add a case for `Action::AutoCompoundDelegatedStake` now that https://github.com/OAK-Foundation/OAK-blockchain/pull/180 is merged?